File: src/main/kotlin/ru/spbau/mit/roguelike/screens/DropScreen.java
package ru.spbau.mit.roguelike.screens;


import ru.spbau.mit.roguelike.entities.Creature;
import ru.spbau.mit.roguelike.world.Item;

public class DropScreen extends InventoryBasedScreen {

    public DropScreen(Creature player) {
        super(player);
    }

    @Override
    protected boolean isAcceptable(Item item) {
        return true;
    }

    @Override
    protected String getVerb() {
        return "drop";
    }

    @Override
    protected Screen use(Item item) {
        player.drop(item);
        return null;
    }
}

File: src/main/kotlin/ru/spbau/mit/roguelike/screens/DropScreen.kt
package ru.spbau.mit.roguelike.screens

import ru.spbau.mit.roguelike.entities.Creature
import ru.spbau.mit.roguelike.world.Item

class DropScreen(player: Creature) : InventoryBasedScreen(player) {

    override fun isAcceptable(item: Item): Boolean {
        return true
    }

    override val verb = "drop"
    override fun use(item: Item): Screen? {
        player.drop(item)
        return null
    }
}
